### PR TITLE
Properly delete visits

### PR DESCRIPTION
**Pull request by 6a68**
@nchapman @pdehaan r?

TL;DR: visit deletion was busted because of the bug fixed by this patch.

The visit.delete model code was expecting an integer count to be returned by postgres, but `{count: <int>}` was getting returned instead. By the laws of javascript insanity, the code then declared shenanigans, told itself that `count > 0` was definitely not true, and, thus, decided that it was time to delete the user_page, when in fact visits still existed that corresponded to said user_page. That led to a postgres error being thrown and unhandled (because the visits table references the user_pages table as a foreign key, and we do _not_ cascade deletes, in order to avoid accidentally nuking data thanks to exactly this kind of bug!). I suspect/hope the DB connection leakage was happening when that uncaught error nuked the process, since systemctl merrily spun up a new one in response.

I've filed #282 to figure out wtf is going on with our error handling cleanup in the models; the problem may simply be that no graceful shutdown happened, so the db connection was leaked. Still seems highly suboptimal handle uncaught exceptions in this way, since, uh, I expect this won't be the last one that leaks :-\

To be continued, I guess.

I've also fixed a typo that was causing jshint to barf.

The bug fix commit fixes the problems we're seeing in the logs on dev, so (fingers crossed) dev should be solid again after landing this.

### CODE REVIEW

**Comment by 6a68** [comment]
I should also mention that this branch is currently on dev. If it's still wonky or busted, maybe I need to dig some more and so we shouldn't land this branch.

**Comment by pdehaan** [comment]
I'm still seeing some issues with deleting items on the dev server. Basically my FF crashed this morning and came back with about 40 results of "New Tab" which got pushed into my Chronicle feed. So I'm trying to go through and delete all those entries manually in Chronicle and my Developer Tools ends up spitting up this message:

> _"Uncaught Error: A `url` property or function must be specified"_

The stack trace is pretty useless, but seems to point to a reference to `t.extend.destroyModel` in [/app/scripts/views/visits/item.js:40](https://github.com/mozilla/chronicle/blob/98d26557937193e8515ccfa267bdb6a905a64ac8/app/scripts/views/visits/item.js#L40)

```js
36:    destroyModel: function (event) {
37:      event.preventDefault();
38:
39:      if (window.confirm('Destroy this visit?')) {
40:        this.model.destroy();
41:
42:        this.$el.fadeOut(function () {
43:          this.remove();
44:        }.bind(this));
45:      }
46:    }
```
Not sure if we need a `try {} catch (err) {}` around that to see why it's sad. Or why it seems to work some times and then begins failing.

chronicle_and_fdc55c2c022172eb5d9c37b9f404fc70_json_ _embedly-cli_and_config_get_txt_ _chronicle

**Comment by pdehaan** [comment]
Long story short, it's probably me who keeps murdering the poor dev server with delete-fest 2014.

**Comment by 6a68** [comment]
@pdehaan That actually makes sense - I've seen a lot of similar bugs lately where Nightly loses the URL when opening a new tab, or detaching a tab from a tab group, or after restarting - sometimes the page loads but the URL is blank, sometimes the page just doesn't load at all. If the addon says, "here you go, server, here are 20 visits to 'about:blank' at the exact same time", then yeah, things are gonna get weird.

**Comment by 6a68** [comment]
Well, let's merge this puppy and see what happens :-)

**Comment by pdehaan** [comment]
Tested in Firefox a few mins ago and I'm seeing this now (when deleting my 20-30 "New Tab" items):
chronicle_and_javascript_-_put_item_on_dynamodb_table_using_aws_sdk_for_node_js_-_stack_overflow

The good news is that they are removed from the DOM after I press delete, and seemingly are still gone after I refresh the page, so it's just a noisy console.

Not specific to "New Tab", as I can reproduce when deleting boring Google search history items or the "Chronicle" page items.
